crates/store/re_dataframe2/src/query.rs

```diff
@@ -527,6 +527,13 @@ impl QueryHandle<'_> {
             .min_by_key(|streaming_state| streaming_state.index_value)
             .map(|streaming_state| streaming_state.index_value)?;
 
+        if let Some(filtered_index_values) = self.query.filtered_index_values.as_ref() {
+            if !filtered_index_values.contains(&cur_index_value) {
+                self.increment_cursors_at_index_value(cur_index_value);
```

Comment:
I know we're not optimizing yet... but it seems like this could take an optional `next_index_value` which we can easily look up  from `filtered_index_values`. Then we could just increment repeatedly in the inner increment as long as we're less than `next_index_value`.

Additionally, if we made `increment_cursors_at_index_value()` return the minimum value across all the chunks, we could loop here and basically guarantee that the call to `next_row()` will end up with a value included in `filtered_index_values`

Reply:
Index sampling and clear/tombstone support first -- I don't want to optimize myself into a corner because of weird interactions with other features...